core/lib/zksync_core/src/metadata_calculator/updater.rs

```diff
@@ -207,10 +207,6 @@ impl TreeUpdater {
                             .await;
                     }
                 }
-                prover_storage
-                    .witness_generator_dal()
-                    .save_witness_inputs(l1_batch_number, object_key, protocol_version_id)
-                    .await;
```

Comment:
Just to understand the remaining use of `prover_storage` in `MetadataCalculator`: It seems that it's only used to sync protocol versions via `.protocol_versions_dal().save_prover_protocol_version(_)`. Is this use still valid? AFAICT, there's a separate `save_prover_protocol_version()` for the FRI prover, and it's appropriately called from the prover system.

Reply:
cc @EmilLuta